libraries/core_libs/network/src/tarcap/packets_handlers/vote_packet_handler.cpp

```diff
@@ -36,22 +100,26 @@ void VotePacketHandler::process(const PacketData &packet_data, const std::shared
                    << ") already seen.";
       continue;
     }
-    peer->markVoteAsKnown(vote_hash);
 
     // Standard vote
     if (vote->getPeriod() >= current_pbft_period) {
       if (vote_mgr_->voteInVerifiedMap(vote)) {
         LOG(log_dg_) << "Vote " << vote_hash.abridged() << " already inserted in verified queue";
-      }
-
-      if (auto vote_is_valid = validateStandardVote(vote); vote_is_valid.first == false) {
-        LOG(log_wr_) << "Vote " << vote_hash.abridged() << " validation failed. Err: " << vote_is_valid.second;
-        continue;
-      }
-
-      if (!vote_mgr_->addVerifiedVote(vote)) {
-        LOG(log_dg_) << "Vote " << vote_hash << " already inserted in verified queue(race condition)";
-        continue;
+        // We don't need to validate and process it, but should add to votes for gossiping
+      } else {
+        if (!shouldProcessVote(vote, peer)) {
```

Comment:
- this should be called inside validateStandardVote as it is check related to standard votes and validateStandardVote is also called in votes sync packet handler and right now you dont call it there even though we should...
- also I still dont get the generic name shouldProcessVote. This is pretty simple and specific function, which is checking max bounds for period, round and step and with proper name I would not need to open that function at all when going through code. When I see shouldProcessVote I have no idea what is going on inside so I have to check it if I am not that familiar with the code

Reply:
If we will also check this inside of `VotesSyncPacketHandler`,  then we wouldn't process votes with bigger period\round\step than our configured values, so for me it looks like syncing won't work in most of the cases 
Thats why it is only here and not inside of this `validateStandardVote`
Is `checkVoteMaxBounds` is good enough you think? Because I don't like something as `checkVoteMaxPeriodRoundStep` is kind of too big